packages/button/src/index.tsx

```diff
@@ -21,66 +23,120 @@ type StyledComponentElement =
   | keyof JSX.IntrinsicElements
   | React.ComponentType<any>
 
-const StyledText = styled(Text)<StyledTextProps>`
+interface ButtonIconAttachmentProps {
+  iconPosition?: ButtonIconPosition
+}
+
+const StyledText = styled(Text)<StyledTextProps & ButtonIconAttachmentProps>`
   border: none;
   cursor: pointer;
   border-radius: ${_ => _.borderRadius}px;
   height: ${_ => _.height}px;
   padding-left: ${_ => Math.round(_.height / 2)}px;
   padding-right: ${_ => Math.round(_.height / 2)}px;
+  margin-left: 0;
+  margin-right: 0;
   vertical-align: middle;
   ${_ =>
     _.full
       ? {
-          width: '100%'
+          width: '100%',
+          display: _.icon ? 'flex' : 'inline-flex'
```

Comment:
Something happened with vertical align of the text (without icons) on Firefox:
<img width="689" alt="Zrzut ekranu 2019-10-29 o 08 32 19" src="https://user-images.githubusercontent.com/8265332/67746610-1a189d00-fa27-11e9-8e9e-fd088e91ee8a.png">
Not sure if caused by those changes, but definitely necessary to be fixed

Reply:
fixed